File: Template Method/PHP/TwoFactorEmail.php
<?php

require_once "TwoFactorAuthentication.php";

class TwoFactorEmail extends TwoFactorAuthentication {
    public function getData(string $username) {
        echo "Getting email data for {$username}\n";
    }

    public function sendNotification(string $username) {
        echo "Sending email to {$username}\n";
    }
    
    public function match(string $username) {
        echo "Matching email code from {$username}\n";
    }
}

File: Template Method/Dart/TwoFactorEmail.dart
import 'TwoFactorAuthentication.dart';

class TwoFactorEmail extends TwoFactorAuthentication {
  @override
  void getData(String username) {
    print("Getting email data for $username");
  }

  @override
  void match(String username) {
    print("Matching email code from $username");
  }

  @override
  void sendNotification(String username) {
    print("Sending email to $username");
  }
}